twenty-first/src/shared_math/tip5.rs

```diff
@@ -909,6 +912,37 @@ mod tip5_tests {
         );
     }
 
+    #[test]
+    fn hash_var_len_equivalence_corner_cases() {
+        // Verify that `hash_var_len` is equivalent
+        // to pad + repeated absorb + squeeze once
+        for preimage_length in [0, 1, 2, 3, 4, 5, 6, 9, 10, 11] {
+            let preimage = random_elements(preimage_length);
+            let hash_varlen_digest = Tip5::hash_varlen(&preimage);
+
+            let mut sponge = Tip5::init();
+            Tip5::pad_and_absorb_all(&mut sponge, &preimage);
+            let squeeze_result = Tip5::squeeze_once(&mut sponge);
+            let digest_through_pad_squeeze_absorb =
+                Digest::new((&squeeze_result[..DIGEST_LENGTH]).try_into().unwrap());
+            assert_eq!(digest_through_pad_squeeze_absorb, hash_varlen_digest);
```

Comment:
These lines shared identically in this test and the test below it. Let's factor them out into a method.